numba/unicode.py

```diff
@@ -540,6 +540,93 @@ def find_impl(a, b):
         return find_impl
 
 
+@register_jitable
+def _check_type(ty, name, accepted=(types.NoneType,)):
+    """
+    Check object belongs to one of specified types
+    :param ty: object type
+    :param name: name of the argument
+    :param accepted: accepted types
+    """
+    thety = ty
+    # if the type is omitted, the concrete type is the value
+    if isinstance(ty, types.Omitted):
+        thety = ty.value
+    # if the type is optional, the concrete type is the captured type
+    elif isinstance(ty, types.Optional):
+        thety = ty.type
+
+    if thety is not None and not isinstance(thety, accepted):
+        raise TypingError('"{}" must be {}, not {}'.format(name, accepted, ty))
+
+
+@register_jitable
+def _adjust_start(length, start=0):
```

Comment:
Are these tiny adjustment functions going to reused in other unicode feature implementations? If not, perhaps just write them as closures in the impl itself https://github.com/numba/numba/pull/4652/files#diff-a21787256ad779b4c4c5c99915c71025R608

Reply:
Ok, let me make the functions as closures.